llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1820,7 +1835,8 @@ bool MemCpyOptPass::processMemCpy(MemCpyInst *M, BasicBlock::iterator &BBI) {
       }
     }
 
-    if (hasUndefContents(MSSA, BAA, M->getSource(), MD, M->getLength())) {
+    if (hadUndefContentsBefore(MSSA, BAA, M->getSource(), AnyClobber, SrcLoc,
+                               M->getLength())) {
```

Comment:
Isn't this going to recompute the clobbering access now, even though we already know it?

Reply:
Yes. I assumed it was inexpensive since it is a caching walker, but I could add a bool parameter if necessary to optimize that